packages/vela/src/Vela.ts

```diff
@@ -0,0 +1,83 @@
+import {
+  type MintActionParams,
+  type TransactionFilter,
+  type SwapActionParams,
+  type StakeActionParams,
+  compressJson,
+} from '@rabbitholegg/questdk'
+import { type Address } from 'viem'
+import { VAULT_CONTRACT, CHAIN_TO_TOKENS, TOKENFARM_CONTRACT, VLP_CONTRACT, VELA_CONTRACT } from './contract-addresses'
+import { CHAIN_ID_ARRAY } from './chain-ids'
+import { TOKENFARM_ABI, TOKENFARM_ABI2, VAULT_ABI } from './abi'
+
+export const swap = async (
+  swap: SwapActionParams,
+): Promise<TransactionFilter> => {
+  const { chainId, contractAddress, tokenIn, amountIn, amountOut, deadline, recipient } = swap
+
+  return compressJson({
+    chainId,
+    to: contractAddress ?? VAULT_CONTRACT,
+    from: recipient,
+    input: {
+      $abi: VAULT_ABI,
+      a: amountIn,
+      b: amountOut, 
+      c: deadline,
+    },
+  })
+}
+
+export const mint = async (
+  mint: SwapActionParams,
```

Comment:
Our mint action is mainly for NFTs and will not be able to work properly with amounts. 

I notice you are using SwapActionParams here, which wont play well with our frontend.

We could just use a swap action for this, but it might be a bit funky in regards to our frontend (The quest would be labeled as "Swap on Vela" and our users would get confused since your UI says Mint).

Reply:
if possible, I hope you can create action type for vela.
Honestly, it doesn't look good to me.
But for following your sdk format, I am just using the actionType.

Reply:
Would just concentrate on the staking action for now, until we figure out a good way to match this action.

Reply:
sure.
thank you!